README.md

```diff
@@ -262,7 +262,239 @@ export default NuxtAuthHandler({
+- [`auth`-key in `nuxt.config.ts`](#nuxtconfigts): Configure the module itself, e.g., where the auth-endpoints are, what origin the app is deployed to, ...
+- [NuxtAuthHandler](#nuxtauthhandler): Configure the authentication behavior, e.g., what authentication providers to use
+
+For a production deployment, you will have to at least set the:
+- `origin` inside the `nuxt.config.ts` config (equivalent to `NEXTAUTH_URL` environment variable),
+- `secret` inside the `NuxtAuthHandler` config (equivalent to `NEXTAUTH_SECRET` environment variable)
+- `STRAPI_BASE_URL` Strapi base URL for all API endpoints by default http://localhost:1337 
+- `STRAPI_API_TOKEN_SECRET` optional, but required for server-to-server and/or headless usage.
+
+1. In your .env file add the following lines:
+```ts
+// Strapi v4 url, out of the box
+ STRAPI_BASE_URL=http://localhost:1337/api
+ STRAPI_API_TOKEN_SECRET=optional
+```
+
+2. In your `nuxt.config.ts` file, configure it like the following
+```ts
+export default defineNuxtConfig({
+  runtimeConfig: {
+    // The private keys which are only available server-side
+    NUXT_SECRET: process.env.NUXT_SECRET,
+    // Default http://localhost:1337/api
+    STRAPI_BASE_URL: process.env.STRAPI_BASE_URL,
+    // Optional but required for server-to-server ops
+    STRAPI_APITOKEN_SECRET:process.env.STRAPI_APITOKEN_SECRET
+  },
+  auth: {
+    origin: process.env.ORIGIN,
+  },
+});
+
+```
+
+3. Create the catch-all nitro route and add the this custom Strapi credentials provider.
+```ts
+// file: ~/server/api/auth/[...].ts
+import CredentialsProvider from "next-auth/providers/credentials";
+import { NuxtAuthHandler } from "#auth";
+
+export default NuxtAuthHandler({
+  secret: process.env.NUXT_SECRET,
+  providers: [
+    // @ts-ignore Import is exported on .default during SSR, so we need to call it this way. May be fixed via Vite at some point
+    CredentialsProvider.default({
+      name: "Credentials",
+      credentials: {
+        // Object is required but can be left empty.
+      },
+      async authorize(credentials: any) {
+        const response = await $fetch(
+          `${process.env.STRAPI_BASE_URL}/auth/local/`,
+          {
+            method: "POST",
+            body: JSON.stringify({
+              identifier: credentials.username,
+              password: credentials.password,
+            }),
+          }
+        );
+
+        if (response.user) {
+          const u = {
+            id: response.id,
+            name: response.user.username,
+            email: response.user.jwt,
+            // Passing the original JWT token through the email field.
+            // IMPORTANT: Always pass encoded JWTs,
+            // don't pass around decoded JWTs.
+          };
+          return u;
+        } else {
+          return null;
+        }
+      },
+    }),
+  ]
+});
+```
+
+###### Passing the headers after authenticating on a protected page
```

Comment:
this is already part of our docs here: https://github.com/sidebase/nuxt-auth#application-side-jwt-token-access

Please remove the section, sorry, but it goes too far out of scope of this PR + adds nothing that is not already there.